libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -873,7 +873,9 @@ std::string YulUtilFunctions::resizeDynamicArrayFunction(ArrayType const& _type)
 					let arrayDataStart := <dataPosition>(array)
 					let deleteStart := add(arrayDataStart, newSlotCount)
 					let deleteEnd := add(arrayDataStart, oldSlotCount)
-					<clearStorageRange>(deleteStart, deleteEnd)
+					let offset := mul(mod(newLen, <itemsPerSlot>), <storageBytes>)
+					if gt(offset, 0) { deleteStart := sub(deleteStart, 1) }
```

Comment:
Without comment, I find this counter-intuitive...

Reply:
Added comment there

Reply:
Plus small refactor that this calculation is done only for packed arrays